chaincfg/chainhash/hashfuncs.go

```diff
@@ -31,3 +34,15 @@ func DoubleHashH(b []byte) Hash {
 	first := sha256.Sum256(b)
 	return Hash(sha256.Sum256(first[:]))
 }
+
+// DoubleHashRaw calculates hash(hash(h)) and returns the resulting bytes.
+func DoubleHashRaw(h hash.Hash) []byte {
```

Comment:
Perhaps specify that the `hash.Hash` instance should already have the message pre-image written to it?

I think with that added context, the test below is also easier to understand.